packages/core/src/components/QueryBuilder/queryToMongo.tsx

```diff
@@ -16,17 +16,16 @@ function isAnd(rule) {
 
 const groupHasNoSubGroupsOrJustOneOR = (group) => {
   let foundOr = false;
-  // eslint-disable-next-line no-plusplus
-  for (let i = 0; i !== group.rules.length; ++i) {
-    const rule = group.rules[i];
+
+  group.rules.forEach((rule) => {
```

Comment:
This is not quite equivalent, because of the `return`. We can leave this as it was - this function is scary

Reply:
Yup, we need to leave this as it was. Applitools broke because of this since the function is not behaving as it was before.